File: python-import-csv/import_data.py
from rushdb import RushDB
import os

api_key = os.getenv('RUSHDB_API_TOKEN')
base_url = os.getenv('RUSHDB_BASE_URL')

def import_data():
    """Import data into RushDB."""
    try:

        db = RushDB(api_key, base_url=base_url)

        with open(os.path.join(os.path.dirname(__file__), 'test_data', 'data.csv'), 'r') as f:
            books_csv = f.read()

        db.records.import_csv(
            label="BOOK",
            data=books_csv
        )

        print("✅ Data imported successfully")
    except Exception as e:
        print(f"❌ Error importing data: {e}")

if __name__ == "__main__":
    import_data()

File: express-books-rag/src/import-data.ts
import path from 'path';
import fs from 'fs-extra';
import { RushDB } from '@rushdb/javascript-sdk';
import { config } from './config.js';
import { parse } from 'csv-parse/sync';

async function importData() {
  try {
    const db = new RushDB(config.rushdbToken, {
      url: config.rushdbBaseUrl,
    });

    const csvPath = path.resolve(process.cwd(), 'test_data', 'data.csv');
    const booksCsv = await fs.readFile(csvPath, { encoding: 'utf-8' });

    const records: any[] = parse(booksCsv, {
      columns: true,
      skip_empty_lines: true,
      trim: true,
    });

    console.log('✅ Parsed JSON:', records);

    await db.records.createMany({
      label: 'BOOK',
      data: records,
      options: { suggestTypes: true, convertNumericValuesToNumbers: true },
    });

    console.log('✅ Data imported successfully');
  } catch (e: any) {
    console.error('❌ Error importing data:', e.message);
    process.exit(1);
  }
}

if (import.meta.url.endsWith('/import-data.ts')) {
  importData().catch((err) => {
    console.error('❌ Error importing data:', err.message);
    process.exit(1);
  });
}
